ncm-network/src/main/perl/nmstate.pm

```diff
@@ -159,6 +159,22 @@ sub make_nm_ip_route
     return \@rt_entry;
 }
 
+# create an absent route entry.
+# if you prepend the routes with the 'absent', then nmstate will clear the existing matches and apply the routes
+# This will allow nmstate to clear all routes for the interface and only apply routes defined in config.
+# useful when routes are changed later on in profile once host is built.
+# return arrayref
+sub make_nm_route_absent {
+    my ($self, $device) = @_;
+
+    my @rt_entry;
+    my %rt;
```

Comment:
This is fine but doesn't feel very Perl-ish. My Perl is rusty but I think you want:

```
my $rt_hf = { 'state' => 'absent', 'next-hop-interface' => $device };
my @rt_entry = ($rt_hf);
```

Or I think you can do it in one:

```
return [ { 'state' => 'absent', 'next-hop-interface' => $device } ];
```

cf. https://stackoverflow.com/questions/17165675/how-to-reference-a-hash-in-perl and https://stackoverflow.com/questions/50155116/create-an-array-ref-in-one-line-in-perl